src/Maestro/SubscriptionActorService/PullRequestDescriptionBuilder.cs

```diff
@@ -0,0 +1,177 @@
+using Maestro.Data.Models;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using static SubscriptionActorService.PullRequestActorImplementation;
+
+namespace SubscriptionActorService
+{
+    public class PullRequestDescriptionBuilder
+    {
+        public PullRequestDescriptionBuilder(ILoggerFactory loggerFactory, StringBuilder description)
```

Comment:
I think this class would have a bit nicer interface if the `StringBuilder` was created internally by this class. I see two issues with the current design:
- This class does not control the `StringBuilder` so other code can do something unexpected with it between the calls to `CalculatePRDescription` like clearing it for example
- For someone reading the code it may look a bit magical because you have to be aware that the output is a side effect of writing to `description` that you passed as constructor argument. If the result was built internally you would have to get it in more explicit way, e.g. (pseudocode):

```csharp
var builder = new PullRequestDescriptionBuilder(loggerFactory);
foreach (var commit in commits)
{
    builder.CalculatePRDescription(...);
}

// New method that just returns stringBuilder.ToString() for private builder
string result = builder.GetDescription();

// then you can append it to other builder
description.Append(result);
```

I think this would make it easier to grasp what is happening when you're reading the code.

Reply:
I think this also lets us make this a static helper class. This class doesn't seem to store any data that it needs per instance. This class' public interface right now is just `CalculatePRDescriptionAsync`, and that method gets all the data it needs not through instance variables, but from external data from the caller so it can be a static method that also takes the existing PR description and the logger as parameters.

Reply:
I don't think this is true anymore, I didn't consider the case where we update an existing PR for which we need to calculate the startingReferenceId based on the previous ones, so the class has data

Reply:
yep, with the startingReferenceId it makes sense to not make it just static. But I think @lpatalas comments about how this class has undocumented side effects in constructor arguments still apply. If I instantiate a PullRequestDescriptionBuilder, I would expect to have to interact with it to get a PullRequestDescription, but instead everything is done to a string builder that I passed to the constructor.

Reply:
I refactored the code a bit by removing the StringBuilders from the outside of PullRequestDescriptionBuilder. The descriptions are now moved around with return values. This should remove the ability for something unexpected to happen to the StringBuilders